builder/transaction_builder.go

```diff
@@ -81,9 +82,21 @@ func (b *TransactionBuilder) AddContextInput(contextInput iotago.Input) *Transac
 	return b
 }
 
-// AddAllotment adds the given allotment to the builder.
-func (b *TransactionBuilder) AddAllotment(allotment *iotago.Allotment) *TransactionBuilder {
-	b.transaction.Allotments = append(b.transaction.Allotments, allotment)
+// IncreaseAllotment adds or increases the given allotment to the builder.
+func (b *TransactionBuilder) IncreaseAllotment(accountID iotago.AccountID, value iotago.Mana) *TransactionBuilder {
+	// check if the allotment already exists and add the value on top
```

Comment:
```suggestion
	if value == 0 {
        return b
    }
```

Reply:
This suggestion came out weird, but I obviously don't meant to delete the comment, just add the check for zero value